packages/esm-patient-chart-app/src/clinical-views/utils/helpers.ts

```diff
@@ -0,0 +1,189 @@
+import { age, formatDate, type OpenmrsResource, parseDate } from '@openmrs/esm-framework';
+import { type TFunction } from 'i18next';
+import { esmPatientChartSchema } from '../../config-schema';
+
+export interface Observation {
+  uuid: string;
+  concept: { uuid: string; name: string };
+  value:
+    | {
+        uuid: string;
+        name: {
+          name: string;
+          display: string;
+        };
+        names?: Array<{ uuid: string; name: string; conceptNameType: string }>;
+      }
+    | string;
+  groupMembers?: Array<Observation>;
+  obsDatetime: string;
+}
+
+export interface Encounter extends OpenmrsResource {
+  encounterDatetime: Date;
+  encounterType: { uuid: string; name: string };
+  patient: {
+    uuid: string;
+    display: string;
+    age: number;
+    birthDate: string;
+  };
+  location: {
+    uuid: string;
+    display: string;
+    name: string;
+  };
+  encounterProviders?: Array<{ encounterRole: string; provider: { uuid: string; name: string } }>;
+  obs: Array<Observation>;
+  form?: {
+    uuid: string;
+  };
+  visit?: string;
+}
+
+export function getEncounterValues(encounter: Encounter, param: string, isDate?: Boolean) {
+  if (isDate) return formatDate(encounter[param]);
+  else return encounter[param] ? encounter[param] : '--';
+}
+
+export function obsArrayDateComparator(left: Observation, right: Observation) {
+  return new Date(right.obsDatetime).getTime() - new Date(left.obsDatetime).getTime();
+}
+
+export function findObs(encounter: Encounter, obsConcept: string): Observation {
+  const allObs = encounter?.obs?.filter((observation) => observation.concept.uuid === obsConcept) || [];
+  return allObs?.length == 1 ? allObs[0] : allObs?.sort(obsArrayDateComparator)[0];
+}
+
+export function getObsFromEncounters(encounters: Encounter, obsConcept: string) {
+  const filteredEnc = encounters?.find((enc) => enc.obs.find((obs) => obs.concept.uuid === obsConcept));
+  return getObsFromEncounter(filteredEnc, obsConcept);
```

Comment:
This is one of the many reasons that I think `getObsFromEncounter` needs refactoring. It's doing too many things with too many concerns.

Reply:
i have seperated some few things, like data that is not fetched from the encounter eg a visit type